wgpu-hal/src/gles/device.rs

```diff
@@ -1179,7 +1181,16 @@ impl crate::Device for super::Device {
     ) -> Result<super::BindGroup, crate::DeviceError> {
         let mut contents = Vec::new();
 
-        for (entry, layout) in desc.entries.iter().zip(desc.layout.entries.iter()) {
+        let layout_and_entry_iter = desc.entries.iter().map(|entry| {
+            let layout = desc
+                .layout
+                .entries
+                .iter()
+                .find(|layout_entry| layout_entry.binding == entry.binding)
+                .expect("internal error: no layout entry found with binding slot");
+            (entry, layout)
+        });
+        for (entry, layout) in layout_and_entry_iter {
```

Comment:
Face-to-face discussion with @jimblandy: This is a quadratic (i.e. `O(n^2)`) algorithm. Not sure if this will be a perf. issue, but it's remarkable here since this is in the path to constructing passes. We should file a separate issue to perf. follow-up.

Reply:
Filed: https://github.com/gfx-rs/wgpu/issues/5473